sktime/transformations/series/kalman_filter/_base.py

```diff
@@ -0,0 +1,270 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Shared code between different KalmanFilterTransformers."""
+
+__author__ = ["NoaWegerhoff", "lielleravid", "oseiskar"]
+
+import numpy as np
+
+
+def _init_matrix(matrices, transform_func, default_val):
```

Comment:
the various private methods are only used by `_kalman_filter.py` - can we move these back to keep everything in one place?

The base class makes sense to have here since imported from multiple places.

Reply:
`_init_matrix` and `_validate_param_shape` are currently used by both `BaseKalmanFilter` and the classes in `_kalman_filter.py`

Reply:
I see - makes sense to keep it there, then.